api/indexer/api.go

```diff
@@ -0,0 +1,81 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package indexer
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/trace"
+
+	"github.com/ava-labs/hypersdk/fees"
+	"github.com/ava-labs/hypersdk/rpc"
+)
+
+var ErrTxNotFound = errors.New("tx not found")
+
+func NewAPIFactory(indexer *TxDBIndexer, name string, path string) *APIFactory {
+	return &APIFactory{
+		path:    path,
+		name:    name,
+		indexer: indexer,
+	}
+}
+
+type APIFactory struct {
+	path    string
+	name    string
+	indexer *TxDBIndexer
+}
+
+func (f *APIFactory) New(vm rpc.VM) (rpc.HTTPHandler, error) {
```

Comment:
So my thoughts are:
1. `New` needs to give us the type to build our api
  - If we want to use generics to get that type, we are effectively going to need to keep the controller vs vm api pattern that we have now to support hypersdk defined abstractions and application defined abstractions.
  - If this package defines its own interface to use (so some interface `indexer.API`), hypersdk can't depend on that package since it would be a circular dependency as the indexer is logically a downstream component from the vm.
2. I think it's reasonable for hypersdk to define a `VM` type (or set of types) that downstream apis can build off of. I think ideally this allows access to current state, the mempool, and subscriptions for consensus events (block acceptance).